### ISSUE TITLE
[Bug]: "Code2Wav input_ids length 1" warning emitted for every request on async_chunk Qwen3-TTS, regardless of caller

### ISSUE THREAD

**Issue by pratapyash** [vllm-project/vllm-omni#3607]
## Summary

vllm-omni 0.20.0 emits the following `WARNING` **on every request** when serving Qwen3-TTS in `async_chunk: true` mode:

```
WARNING [qwen3_tts_code2wav.py:156] Code2Wav input_ids length 1 not divisible by num_quantizers 16; skipping malformed request.
```

The warning is **benign** -- the request still finishes and previously-decoded audio chunks are returned correctly -- but it produces high log volume in production and the "malformed request" wording falsely implies a client error.

## Reproduction

Any caller of `AsyncOmni.generate()` on a Qwen3-TTS-12Hz model with `async_chunk: true` reproduces it. Verified across:

- `vllm-omni serve --omni Qwen/Qwen3-TTS-12Hz-1.7B-CustomVoice` (CLI subprocess hitting `/v1/audio/speech`)
- `OmniOpenAIServingSpeech.create_speech` reused in-process
- Direct `AsyncOmni.generate(...)` from a plain Python script

The warning fires per request in all three. None of the following caller-side variations changed that:

- `tokens_input([1]*ph_len)` vs plain dict prompt construction
- `coerce_param_message_types(list(defaults), is_streaming=True)` vs handrolled `output_kind=DELTA`
- `stop_token_ids: [2150]` on stage-0 default sampling params
- Adding `voice_created_at: [0]` to `additional_information`

## Why the warning exists (what it actually validates)

`num_quantizers=16` is the codebook count of the Qwen3-TTS-12Hz residual vector quantizer (RVQ): each codec frame is encoded as 16 parallel codebook indices, one per quantizer level. Code2Wav's `forward()` receives a flat 1-D tensor `input_ids` whose length must be `num_frames x num_quantizers` so the codes can be reshaped into `[num_frames, 16]` and fed to the decoder. The divisibility check at `qwen3_tts_code2wav.py:156` enforces this invariant -- a length not divisible by 16 means the input doesn't represent a whole number of frames, so the decoder cannot produce a coherent waveform and the request is skipped (an empty audio buffer is appended). The warning is correct **as a defensive guard**: a length-1 input is genuinely malformed for code2wav. The bug is upstream of the guard -- the pipeline is producing a length-1 input on every request via the EOS pad-zero injection (traced below), which the guard then catches.

## Root cause trace

End-of-stream pad-zero injection at the talker -> code2wav handoff. File:line at v0.20.0:

1. **Talker finishes between codec-chunk boundaries.** `_extract_last_frame` (`stage_input_processors/qwen3_tts.py:127-128`) rejects the all-zero EOS frame; no frame is appended.

2. **Empty-codes-finish payload.** `talker2code2wav_async_chunk` (`stage_input_processors/qwen3_tts.py:209-214`) sees `length <= 0 and finished` and emits:
   ```python
   {"codes": {"audio": []}, "meta": {"finished": True}}
   ```

3. **Receiver writes empty `prompt_token_ids`.** `chunk_transfer_adapter.py:165-166` sets `request.prompt_token_ids = []` and adds the request to `finished_requests`.

4. **Scheduler pad-zero injection.** `OmniGenerationScheduler.schedule()` (`core/sched/omni_generation_scheduler.py:95-105`, `:150-156`) sees `len(prompt_token_ids) == 0` for a finished request and injects a single `0` so the stage can emit a FINISHED `EngineCoreOutput`.

5. **Code2Wav rejects the 1-token input.** `Qwen3TTSCode2Wav.forward()` (`model_executor/models/qwen3_tts/qwen3_tts_code2wav.py:147-162`) sees `input_ids = tensor([0])`, fails `n % num_quantizers == 0` (1 is not a multiple of 16), logs the warning, sets `audios[idx] = torch.zeros(0)`, continues.

Audio output is correct because earlier valid chunks have already been decoded; only this terminal forward pass is wasted.

## Related

- #3124 -- exact warning title, closed `invalid`. Different surface cause (Base voice-clone with mismatched ref-text).
- #2911 -- moved `RequestOutputKind.DELTA` enforcement out of `AsyncOmni.generate` into the entrypoint. Required by direct `AsyncOmni.generate` callers; unrelated to this warning.
- #3396 -- analogous Qwen2.5-Omni fix defensively skipping empty codec input updates. Same shape of fix, different model.

## Environment

- vllm-omni: 0.20.0 (PyPI)
- vllm: 0.20.2
- Model: `Qwen/Qwen3-TTS-12Hz-1.7B-CustomVoice` (also reproduced on `0.6B-CustomVoice`)
- Deploy YAML: bundled `vllm_omni/deploy/qwen3_tts.yaml`
- GPU: NVIDIA L40S (sm_89)

**Comment by hsliuustc0106** [comment]
i also met this problem
